### PR TITLE
Add usr/bin to the shebang replace path

### PR DESCRIPTION
**Pull request by sergiusens**
LP: #1534812

### CODE REVIEW

**Comment by kyrofa** [comment]
:+1: from me, though we may want to do this to all .debs in the future.

**Comment by kyrofa** [comment]
Ah right, the downside to doing it this way: It also needs to be done to the roscore plugin or the file contents differ.

**Comment by sergiusens** [comment]
I am thinking of rebasing with #236

**Comment by kyrofa** [comment]
Does #236 look good? Feel free to merge and go ahead and rebase.

File: examples/ros/snapcraft.yaml [hunk 1]
```diff
@@ -22,6 +22,28 @@ apps:
...
+        - usr/lib/python2.7/pdb.py
+        - usr/lib/python3.4/pdb.py
+        - usr/share/dh-python/dh_pypy
+        - usr/share/dh-python/dh_python3
+        - usr/share/dh-python/pybuild
+        - usr/share/python3/py3versions.py
+    stage:
+      - -$dev
+    snap:
+      - -$dev
```
Comment (elopio) [inline examples/ros/snapcraft.yaml]: This makes ROS an awesome example. Cool.

File: examples_tests/tests.py [hunk 2]
```diff
@@ -178,6 +178,9 @@ class TestSnapcraftExamples(testscenarios.TestWithScenarios):
             'dir': 'ros',
             'name': 'ros-example',
             'version': '1.0',
+            'external_tests_commands': [
+                ("sed -n '/env/p;1q' snap/usr/bin/rosversion",
+                 b'#!/usr/bin/env python\n',)],
```
Comment (elopio) [inline examples_tests/tests.py]: how clever this guy who invented the external test commands ;)
I would explain in a comment what that sed statement is doing there.

File: snapcraft/repo.py [hunk 3]
```diff
@@ -298,6 +308,14 @@ def _fix_filemode(path):
         os.chmod(path, mode & 0o1777)
 
 
+def _fix_shebangs(path):
```
Comment (elopio) [inline snapcraft/repo.py]: It wouldn't hurt to comment why the shebangs have to be fixed. Just like on fix_symlinks above.

**Comment by elopio** [comment]
+1 from me. I'm just missing a couple of comments.

File: snapcraft/repo.py (hunk 3 above)
Comment (sergiusens) [inline snapcraft/repo.py]: done

File: examples_tests/tests.py (hunk 2 above)
Comment (sergiusens) [inline examples_tests/tests.py]: done

**Comment by sergiusens** [comment]
@elopio I fixed the code to take into account your comments :smile:

**Comment by sergiusens** [comment]
@kyrofa done

**Pull request merged**